### PR TITLE
Added logger (pino) and removed debug

### PR DESCRIPTION
**Pull request by mirpo**
Screenshot 2020-09-09 at 13 48 30

In dev we use `pretty` format and production we'll store NDJSON.
Docs:

- **pretty logs** https://getpino.io/#/docs/pretty?id=pretty-printing#/docs/pretty?id=pretty-printing#/docs/pretty?id=pretty-printing
- **filtering logs** https://getpino.io/#/docs/help?id=log-filtering#/docs/help?id=log-filtering#/docs/help?id=log-filtering
- **transports** (for example pino-sentry, pino-cloudwatch) https://getpino.io/#/docs/transports?id=known-transports#/docs/transports?id=known-transports#/docs/transports?id=known-transports

### CODE REVIEW

File: src/http/DataProduceEndpoints.js [hunk 1]
```diff
@@ -82,6 +86,7 @@ module.exports = (streamFetcher, publisher, partitionFn = partition) => {
                 }
                 signatureType = req.query.signatureType ? parsePositiveInteger(req.query.signatureType) : 0
             } catch (err) {
+                logger.error(err.message)
```
Comment (timoxley) [inline src/http/DataProduceEndpoints.js:89]: You sure you don't want to log the `err` or `err.stack`? Might find it tricky to debug without a stack trace.

File: src/storage/MicroBatchingStrategy.js [hunk 2]
```diff
@@ -27,7 +29,7 @@ class SharedContext {
         } catch (e) {
             const key = `${streamMessages[0].getStreamId()}::${streamMessages[0].getStreamPartition()}`
             if (this.logErrors) {
-                console.error(`Failed to insert (${key}): ${e.stack ? e.stack : e}`)
+                logger.error(`Failed to insert (${key}): ${e.stack ? e.stack : e}`)
```
Comment (timoxley) [inline src/storage/MicroBatchingStrategy.js:32]: 🙀  who is throwing non-errors?

File: src/StreamFetcher.js [hunk 3]
```diff
@@ -70,11 +70,11 @@ module.exports = class StreamFetcher {
...
         }).catch((e) => {
-            console.error(`failed to communicate with E&E: ${e}`)
+            logger.error(`failed to communicate with E&E: ${e}`)
             throw e
         }).then((response) => {
             if (response.status !== 200) {
-                debug(
+                logger.debug(
                     'fetch failed with status %d for streamId %s, apiKey %s, sessionToken %s : %o',
                     response.status, streamId, apiKey, sessionToken, response.text(),
```
Comment (timoxley) [inline src/StreamFetcher.js:79]: not related to this PR but `response.text()` will be a promise won't it

Comment (harbu) [reply src/StreamFetcher.js:79]: Yeah might want to `await`

**Review by harbu — APPROVED** [review]

File: src/storage/MicroBatchingStrategy.js (hunk 2 above)
Comment (mirpo) [reply src/storage/MicroBatchingStrategy.js:32]: 🥇

File: src/http/DataProduceEndpoints.js (hunk 1 above)
Comment (mirpo) [reply src/http/DataProduceEndpoints.js:89]: Updated

File: src/StreamFetcher.js (hunk 3 above)
Comment (mirpo) [reply src/StreamFetcher.js:79]: Updated

**Pull request merged**